hydromt/region/region.py

```diff
@@ -0,0 +1,85 @@
+"""A class to handle all functionalities to do with model regions."""
+
+from logging import Logger, getLogger
+from os.path import exists
+from pathlib import Path
+from typing import Any, Dict, Optional
+
+import numpy as np
+from geopandas import GeoDataFrame, GeoSeries
+
+from hydromt._typing.model_mode import ModelMode
+from hydromt._typing.type_def import StrPath
+from hydromt.region._specifyers import RegionSpecifyer
+
+logger = getLogger(__name__)
+
+
+class Region:
+    """A class to handle all operations to do with the model region."""
+
+    def __init__(
+        self,
+        region_dict: Dict[str, Any],
+        logger: Logger = logger,
+    ):
+        self._dict: Dict[str, Any] = region_dict
+        self._data: Optional[GeoDataFrame] = None
+        self._spec = self._parse_region(region_dict, logger=logger)
+        self.logger = logger
+
+    def write(self, path: StrPath, mode: ModelMode = ModelMode.WRITE):
```

Comment:
can do but that is going to make the testing much more convoluted since then we can't test region and model separately anymore. (again moving to having only integration tests) is this really necessary? I'll make sure that it's always set correctly when used from the model.